ci/do_ci.sh

```diff
@@ -148,6 +148,19 @@ elif [[ "$1" == "bazel.dev" ]]; then
   echo "Building and testing..."
   bazel test ${BAZEL_TEST_OPTIONS} -c fastbuild //test/...
   exit 0
+elif [[ "$1" == "bazel.compiletimeoptions" ]]; then
+  # Right now, none of the available compile-time options conflict with each other. If this
+  # changes, this build type may need to be broken up.
+  COMPILE_TIME_OPTIONS="--define=signal_trace=disabled --define hot_restart=disabled --define google_grpc=disabled"
+  setup_clang_toolchain
+  # This doesn't go into CI but is available for developer convenience.
+  echo "bazel with different compiletime options build with tests..."
+  cd "${ENVOY_CI_DIR}"
+  echo "Building..."
+  bazel build ${BAZEL_BUILD_OPTIONS} ${COMPILE_TIME_OPTIONS} -c fastbuild //source/exe:envoy-static
```

Comment:
Could we use `-c dbg` here, so that we have at least some coverage for it (see #2566)?

Reply:
Good idea